### DISCUSSION TITLE
Container Lifetime for CosmosDB and Azure Storage Account (v9)

### DISCUSSION THREAD

**Discussion by obiwanjacobi** [microsoft/aspire#6774] [Q&A] [answered]
I read about the new `Lifetime()` api to have persistent containers. I wanted to try that out on my CosmosDB and Azure Storage Account **emulators** that run in containers when I debug.

After some investigation the problem seems to be that the `Resource` type that the `AddAzureCosmosDB()` and `AddAzureStorage()` return is not of the correct 'container' type for the `Lifetime()` extension method to be available.

Is there any way to get these containers to be persistent (especially the CosmosDB one that takes a long time to fully start)?

Running dotnet9 and Aspire v9.

**Comment by rudiv** [comment, marked as answer]
Aspire 9 changed some of the signatures slightly.

```csharp
var storage = builder.AddAzureStorage("storage")
    .RunAsEmulator(o =>
    {
        o.WithLifetime(ContainerLifetime.Persistent);
    // ...

var cosmos = builder.AddAzureCosmosDB("nosql")
    .RunAsEmulator(o =>
    {
        o.WithLifetime(ContainerLifetime.Persistent);
    // ...
```
